src/data/markdown/docs/07 extensions/03 Guides/build-k6-using-docker.md

```diff
@@ -0,0 +1,123 @@
+```
+
+</CodeGroup>
+
+This creates a `k6` (or `k6.exe`) binary in the current working directory.
+
+## Breaking down the command
+
+The example command line may look a bit intimidating at first, but let's focus on the first part, which pertains strictly to Docker:
+
+```bash
+docker run --rm -u "$(id -u):$(id -g)" -v "${PWD}:/xk6"
+```
+
+This tells Docker to run a new container from an image. 
+- `--rm` means the container will be destroyed once your build is completed.
+- `-u` specifies the user and group IDs of the account on the host machine. This is important for the `k6` file to have the same file permissions as the host user.
+- `-v` is required to mount the current working directory inside the container, so that the `k6` binary can be written to it.
+
+For Windows and Mac, we additionally include the target system as an environment variable:
+
+```bash
+-e GOOS=<target os>
+```
+
+The remainder is straight from the [xk6 documentation](https://github.com/grafana/xk6/#command-usage), with the exception that we use the `grafana/xk6` _image_ rather than a local installation of `xk6`:
+
+```plain
+grafana/xk6 build [<k6_version>]
+    [--output <file>]
+    [--with <module[@version][=replacement]>...]
+    [--replace <module=replacement>...]
+
+Flags:
+  --output   specifies the new binary name [default: 'k6']
+  --replace  enables override of dependencies for k6 and extensions [default: none]
+  --with     the extension module to be included in the binary [default: none]
+```
+
+> For this portion, use the [interactive builder](/extensions/get-started/bundle/) and prefix the `grafana/` to avoid typing mistakes!
+
+<Blockquote mod="attention">
+
+The use of `--replace` should be considered an advanced feature to be avoided unless required.
+
+</Blockquote>
+
+Referring back to our executed command, note that:
+- We specify the version as `v0.43.1`. When you omit the version or specify `latest`, you build using the _latest_ source code for k6.
+  Consider using a stable [release version](https://github.com/grafana/k6/releases) as a best practice unless you genuinely want the _bleeding edge_.
+- We specify a full GitHub URI for the extension repository with each `--with`.
+If a version is not specified, the default is again the `latest`.
+  Check your extension repository for stable release versions, if available, to lock in your version as we've done with `xk6-kafka@v0.17.0` and `xk6-output-influxdb@v0.3.0`.
+- For Windows, we used the `--output` option to name our result as `k6.exe`; if not specified, our new binary is `k6` within the current directory.
+If you specify a directory, the new binary will be `k6` within _that_ directory.
+If you specify a path to a non-existent file, e.g. `/tmp/k6-extended`, this will be the path and filename for the binary.
+
+Run `./k6 version` (or `k6.exe version`) to check that your build is based on the appropriate `k6` version and contains the desired extensions. For example:
+
```

Comment:
Missing a block code with the output, I don't know how to suggest that 😞 

```
k6 v0.43.1 ((devel), go1.20.1, darwin/amd64)
Extensions:
  github.com/grafana/xk6-output-influxdb v0.3.0, xk6-influxdb [output]
  github.com/mostafa/xk6-kafka v0.17.0, k6/x/kafka [js]
```

Reply:
Oops.